### PR TITLE
pw3270: init at 5.4

### PR DESCRIPTION
**Pull request by vifino**
## Description of changes

pw3270 is an IBM 3270/tn3270 emulator.

It took *a lot* of effort to get this patched enough to build it.
It makes a lot of pretty bad assumptions about the system that just explode spectacularly with Nix.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/li/libv3270/package.nix [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    touch scripts/config.rpath
+    NOCONFIGURE=1 sh ./autogen.sh
+  '';
+
+  enableParallelBuilds = true;
+
+  meta = with lib; {
+    description = "3270 Virtual Terminal for GTK";
+    homepage = "https://github.com/PerryWerneck/libv3270";
+    license = licenses.lgpl3;
```
Comment (Lurkki14) [inline pkgs/by-name/li/libv3270/package.nix:55]:
```suggestion
    license = licenses.lgpl3Plus;
```
Just `licenses.lgpl3` is deprecated

File: pkgs/by-name/li/lib3270/package.nix [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  preConfigure = ''
+    NOCONFIGURE=1 sh autogen.sh
+  '';
+
+  enableParallelBuilds = true;
+
+  meta = with lib; {
+    description = "TN3270 client Library";
+    homepage = "https://github.com/PerryWerneck/lib3270";
+    license = licenses.lgpl3;
```
Comment (Lurkki14) [inline pkgs/by-name/li/lib3270/package.nix:53]:
```suggestion
    license = licenses.lgpl3Plus;
```

File: pkgs/by-name/pw/pw3270/package.nix [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    mv $out/share/gsettings-schemas/${pname}-${version}/glib-2.0/schemas $out/share/glib-2.0/
+    rm -rf $out/share/gsettings-schemas
+  '';
+
+  enableParallelBuilds = true;
+
+  meta = with lib; {
+    description = "3270 Emulator for gtk";
+    homepage = "https://softwarepublico.gov.br/social/pw3270/";
+    license = licenses.lgpl3;
```
Comment (Lurkki14) [inline pkgs/by-name/pw/pw3270/package.nix:72]:
```suggestion
    license = licenses.lgpl3Plus;
```

File: pkgs/by-name/pw/pw3270/package.nix [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    maintainers = [ maintainers.vifino ];
```
Comment (Lurkki14) [inline pkgs/by-name/pw/pw3270/package.nix:73]: `meta.mainProgram` should be set

**Review by Lurkki14 — CHANGES_REQUESTED** [review]

File: pkgs/by-name/li/libv3270/package.nix (hunk 1 above)
Comment (vifino) [reply pkgs/by-name/li/libv3270/package.nix:55]: But it isn't lgpl 3 or later, it's  lgpl 3 only. Are you sure about this?

Comment (Lurkki14) [reply pkgs/by-name/li/libv3270/package.nix:55]: At least here the header mentions LGPL3 or later https://github.com/PerryWerneck/libv3270/blob/master/src/include/terminal.h

Comment (vifino) [reply pkgs/by-name/li/libv3270/package.nix:55]: Huh. Licensed differently all over the place.

I'll change it to lgpl3Plus as that does seem to be the most recently used license in the project.

Thanks!

**Review by Lurkki14 — APPROVED** [review]

**Pull request merged**